Update PULL_REQUEST_TEMPLATE.md (#5035)
cylc/cylc-flow: .github/PULL_REQUEST_TEMPLATE.md

<!--
Thanks for your contribution:

> * Please list any related issues with a "closes" or "addresses" tag.

Author: ```suggestion
* Please list any related issues with a "closes" or "addresses" tag.
* Please add a helpful description.
```

* For bugfixes we have maintenance branches e.g. `8.0.x`, please raise separate
  pull requests against master and the maintenance release branches as appropriate.

> -->

Reviewer: Worth adding "please provide a description"? All too often we're having PRs that just say "closes X" and it takes a fair bit of reading of the original issue + comments to figure out what the change is supposed to do.

Reviewer 2: Agreed!

**Check List**

- [ ] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [ ] Contains logically grouped changes (else tidy your branch by rebase).
- [ ] Does not contain off-topic changes (use other PRs for other changes).
- [ ] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.

> - [ ] Tests are included if required.

Reviewer 2: ```suggestion
- [ ] Tests are included (if not, please explain why they are not needed).
```

> - [ ] Changelog is included if required.

Reviewer 2: Explain when they are required?
```suggestion
- [ ] Changelog is included if this is a significant user-facing change
```

- [ ] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.

> - [ ] [if bugfix] PRs raised to both master and the relevant bugfix branch.

Reviewer 2: ```suggestion
- [ ] If this is a bug fix, PRs raised to both master and the relevant maintenance branch.
```

<!--
Thanks for your contribution:
* Please list any related issues with a "closes" or "addresses" tag.
* Please add a helpful description.
* For bugfixes we have maintenance branches e.g. `8.0.x`, please raise separate
  pull requests against master and the maintenance release branches as appropriate.
-->

**Check List**

- [ ] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [ ] Contains logically grouped changes (else tidy your branch by rebase).
- [ ] Does not contain off-topic changes (use other PRs for other changes).
- [ ] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [ ] Tests are included (or explain why tests are not needed).

> - [ ] `CHANGES.md` entry included if this is a significant change

Reviewer: ```suggestion
- [ ] `CHANGES.md` entry included if this is a change that can affect users
```
